main.js

```diff
@@ -1,58 +1,66 @@
 const bookList = document.querySelector('ul.books');
 const addForm = document.querySelector('form.add-book');
 const storageDetails = localStorage.getItem('books');
-let bookDetails = [];
-
-// Add books function
-function addBook(bookName, author) {
-  const book = {};
-  book.bookName = bookName;
-  book.author = author;
-  bookDetails.push(book);
-  localStorage.setItem('books', JSON.stringify(bookDetails));
-  const liHtml = `<li class="books">${bookName} ${author}<button class="delete">Delete</button></li>`;
-  bookList.insertAdjacentHTML('beforeend', liHtml);
-}
 
-// Delete book function
-function deleteBook(e) {
-  const { target } = e;
-  const parent = e.target.parentElement;
-  if (target.classList.contains('delete')) {
-    const bookName = parent.childNodes[0].textContent;
+// Books class
+
+class Books {
+  constructor(storedBooks) {
+    this.bookDetails = [];
+    // Loading books if exists in local storage
+    if (storedBooks !== null) {
+      this.bookDetails = JSON.parse(storageDetails);
+      this.displayBooks();
+    }
+  }
+
+  // Display books function
+  displayBooks() {
+    let liHtml = '';
+    this.bookDetails.forEach((e) => {
+      liHtml += `<li class="books">"${e.bookName}" by ${e.author}<button class="delete">Remove</button></li>`;
+    });
+    bookList.innerHTML = liHtml;
+  }
 
-    // Removing book from the array
-    bookDetails = bookDetails.filter((e) => bookName !== `${e.bookName} ${e.author}`);
-    // Updating local storage
-    localStorage.setItem('books', JSON.stringify(bookDetails));
+  // Add books function
+  addBook(bookName, author) {
+    const book = {};
+    book.bookName = bookName;
+    book.author = author;
+    this.bookDetails.push(book);
```

Comment:
# Optional:
- You have done a nice job, congratulations. I would advise you to use the method that doesn't mutate the array, e.g. use [concat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) instead of [push](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push).
[See code example](https://www.educative.io/answers/spread-operator-vs-arrayprototypeconcat-in-javascript)